### ISSUE TITLE
[TASK] Replace depecated `fetchAll()` method calls.

### ISSUE THREAD

**Issue by dkd-kaehm** [TYPO3-Solr/ext-solr#2799]
**What should be done in the scope of this task?**

The method `fetchAll()` is deprecated in doctrine. All usages must be replaced with `fetchAllNumeric()`, `fetchAllAssociative()` or `fetchFirstColumn()` accordingly usage.

**Comment by georgringer** [comment]
@dkd-kaehm *DON'T do this. this is only true if doctrine/dbal:^2.12 is used but in 10 both is allowed. With TYPO3 11  this will be required (see https://github.com/TYPO3/TYPO3.CMS/commit/33e9fce5376b232425d72d37f3eb58fe60491ba4#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34) but NOT for TYPO3 which would be then highly breaking!

**Comment by dkd-kaehm** [comment]
Yes, that is for far milestone, if we are closer to TYPO3 11 LTS.
